code/modules/mining/ores_coins.dm

```diff
@@ -39,22 +39,23 @@
 		var/mob/living/silicon/robot/R = AM
 		OB = locate(/obj/item/storage/bag/ore) in R.held_items
 	if(OB)
+		GET_COMPONENT_FROM(STR, /datum/component/storage, OB)

```

Comment:
the abstraction here is much leakier, is there anyway to improve it such that you can just throw a signal here and pass the AM?